### ISSUE TITLE
Entry level bugs in the main script, it cannot be run

### ISSUE THREAD

**Issue by ywu36** [acbull/LADIES#5]
Dear authors,

Thanks for sharing the code, after try to run it, I found multiple entry level bugs in the code. Now the main script is not even executable. Could you help to solve the issues?

For example:
1.  The "row_normalize" function is not even implemented.
2. The "load_data" function version is not correct, it outputs "degrees" in the function, but never been used in "pytorch_ladies.py". The corresponding data holder name is "labels".
3. Function "F.cross_entropy" never takes a matrix containing one hot vectors as ground-truth, the correct ground-truth label should be saved in a int vector.

After I fixed all the issues in the code, the algorithm cannot converge in any of the databases. The training error goes to 0 after 2 epochs but the validation error keep increasing, the F-1 score always fluctuates around 0.7, and the first iteration has the lowest validation loss, which is very confusing. 

Could you please fix the errors in the code? It may mislead the community. Thank you very much!

YH

**Comment by ywu36** [comment]
Loss: 

Epoch: 0 (0.5s) Train Loss: 1.33    Valid Loss: 1.11 Valid F1: 0.572
Epoch: 1 (0.9s) Train Loss: 0.15    Valid Loss: 1.51 Valid F1: 0.746
Epoch: 2 (1.3s) Train Loss: 0.07    Valid Loss: 1.75 Valid F1: 0.738
Epoch: 3 (1.7s) Train Loss: 0.02    Valid Loss: 1.95 Valid F1: 0.748
Epoch: 4 (2.1s) Train Loss: 0.03    Valid Loss: 2.27 Valid F1: 0.702
Epoch: 5 (2.4s) Train Loss: 0.04    Valid Loss: 2.12 Valid F1: 0.738
Epoch: 6 (2.9s) Train Loss: 0.02    Valid Loss: 2.14 Valid F1: 0.726
Epoch: 7 (3.2s) Train Loss: 0.03    Valid Loss: 2.02 Valid F1: 0.756
Epoch: 8 (3.6s) Train Loss: 0.03    Valid Loss: 2.24 Valid F1: 0.726
Epoch: 9 (4.0s) Train Loss: 0.02    Valid Loss: 2.18 Valid F1: 0.742
Epoch: 10 (4.5s) Train Loss: 0.02    Valid Loss: 2.50 Valid F1: 0.708
Epoch: 11 (4.9s) Train Loss: 0.03    Valid Loss: 2.47 Valid F1: 0.722
Epoch: 12 (5.4s) Train Loss: 0.01    Valid Loss: 2.56 Valid F1: 0.716
Epoch: 13 (5.8s) Train Loss: 0.03    Valid Loss: 2.63 Valid F1: 0.734
Epoch: 14 (6.2s) Train Loss: 0.03    Valid Loss: 2.47 Valid F1: 0.720
Epoch: 15 (6.7s) Train Loss: 0.04    Valid Loss: 2.49 Valid F1: 0.730
Epoch: 16 (7.1s) Train Loss: 0.02    Valid Loss: 2.74 Valid F1: 0.696
Epoch: 17 (7.5s) Train Loss: 0.03    Valid Loss: 2.74 Valid F1: 0.684
Epoch: 18 (8.0s) Train Loss: 0.02    Valid Loss: 2.65 Valid F1: 0.722
Epoch: 19 (8.4s) Train Loss: 0.02    Valid Loss: 2.67 Valid F1: 0.710
Epoch: 20 (8.8s) Train Loss: 0.02    Valid Loss: 2.74 Valid F1: 0.700
Epoch: 21 (9.3s) Train Loss: 0.01    Valid Loss: 2.54 Valid F1: 0.722

**Comment by acbull** [comment]
Hi:

Sorry for the trouble. The previous version of code indeed contains many typo bugs. Sorry about that.

I've fixed all the mentioned typos, and you can try the current version.

The results running on PubMed by
python3 pytorch_ladies.py --cuda 0 --dataset pubmed --sample_method ladies --samp_num 512

and the result is:


Epoch: 0 (0.1s) Train Loss: 1.09    Valid Loss: 1.08 Valid F1: 0.480
Epoch: 1 (0.3s) Train Loss: 0.95    Valid Loss: 0.75 Valid F1: 0.612
Epoch: 2 (0.5s) Train Loss: 0.44    Valid Loss: 0.84 Valid F1: 0.728
Epoch: 3 (0.6s) Train Loss: 0.14    Valid Loss: 1.17 Valid F1: 0.778
Epoch: 4 (0.8s) Train Loss: 0.06    Valid Loss: 1.25 Valid F1: 0.796
Epoch: 5 (0.9s) Train Loss: 0.07    Valid Loss: 1.41 Valid F1: 0.804
Epoch: 6 (1.1s) Train Loss: 0.04    Valid Loss: 1.36 Valid F1: 0.774
Epoch: 7 (1.2s) Train Loss: 0.08    Valid Loss: 1.41 Valid F1: 0.792
Epoch: 8 (1.4s) Train Loss: 0.03    Valid Loss: 1.32 Valid F1: 0.794
Epoch: 9 (1.5s) Train Loss: 0.04    Valid Loss: 1.55 Valid F1: 0.794
Epoch: 10 (1.7s) Train Loss: 0.05    Valid Loss: 1.43 Valid F1: 0.774
Epoch: 11 (1.9s) Train Loss: 0.03    Valid Loss: 1.59 Valid F1: 0.792
Epoch: 12 (2.0s) Train Loss: 0.04    Valid Loss: 1.84 Valid F1: 0.778
Epoch: 13 (2.2s) Train Loss: 0.05    Valid Loss: 1.38 Valid F1: 0.766
Epoch: 14 (2.3s) Train Loss: 0.01    Valid Loss: 1.64 Valid F1: 0.780
Epoch: 15 (2.5s) Train Loss: 0.01    Valid Loss: 1.41 Valid F1: 0.780
Epoch: 16 (2.6s) Train Loss: 0.02    Valid Loss: 1.61 Valid F1: 0.766
Epoch: 17 (2.8s) Train Loss: 0.02    Valid Loss: 1.86 Valid F1: 0.770
Epoch: 18 (2.9s) Train Loss: 0.03    Valid Loss: 1.85 Valid F1: 0.774
Epoch: 19 (3.1s) Train Loss: 0.03    Valid Loss: 2.08 Valid F1: 0.758
Epoch: 20 (3.2s) Train Loss: 0.00    Valid Loss: 1.94 Valid F1: 0.760
Epoch: 21 (3.4s) Train Loss: 0.03    Valid Loss: 2.18 Valid F1: 0.782
Epoch: 22 (3.5s) Train Loss: 0.02    Valid Loss: 1.90 Valid F1: 0.748
Epoch: 23 (3.7s) Train Loss: 0.01    Valid Loss: 1.80 Valid F1: 0.776
Epoch: 24 (3.9s) Train Loss: 0.00    Valid Loss: 1.86 Valid F1: 0.762
Iteration: 4, Test F1: 0.791

The validation loss indeed increase, and I guess it's mainly because of overfitting (the best model achieves at epoch 5)
